core/controllers/moderator.py

```diff
@@ -73,12 +73,8 @@ def get(self):
     @acl_decorators.can_access_moderator_page
     def post(self):
         """Handles POST requests."""
-        featured_activity_reference_dicts = self.normalized_payload.get(
+        featured_activity_references = self.normalized_payload.get(
```

Comment:
Same problem; variable name and payload key don't match. Please be very clear about what is being described at each point; do not use dicts to name domain objects and vice versa.

Please fix elsewhere if needed.

Reply:
Thank you for review, will take care of these points while creating another PR.